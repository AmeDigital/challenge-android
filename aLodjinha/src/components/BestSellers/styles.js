import { StyleSheet } from 'react-native';
import MaterialDesign from '../../utils/MaterialDesign';

export default StyleSheet.create({
	container: {
		flex: 1
	},
	bestHeader: {
		paddingHorizontal: 15,
		paddingVertical: 15,
		borderBottomWidth: 1,
		borderColor: MaterialDesign.colorGreyish
	},
	bestTitle: {
		fontSize: 18,
		color: MaterialDesign.colorDark,
		fontWeight: 'bold'
	},
	bestContent: {}
});
