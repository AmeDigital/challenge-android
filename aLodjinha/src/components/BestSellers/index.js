import React, { useState, useEffect } from 'react';
import { Text, View, ScrollView } from 'react-native';

import styles from './styles';
import Product from '../Product';
import api from '../../services/api';
import Loading from '../Loading';

export default function BestSellers({ navigation }) {
	const [ best, setBest ] = useState([]);
	useEffect(() => {
		(async function loadBestSellers() {
			const response = await api.get('/produto/maisvendidos');
			setBest(response.data.data);
		})();
	}, []);

	function funcPage(id, descricao) {
		navigation.navigate('DescriptionProduct', { id, descricao });
	}

	return (
		<View style={styles.container}>
			<View style={styles.bestHeader}>
				<Text style={styles.bestTitle}>Mais Vendidos</Text>
			</View>
			<View>
				{best.length > 0 ? (
					best.map((item) => <Product key={item.id} item={item} funcPage={funcPage} />)
				) : (
					<Loading />
				)}
			</View>
		</View>
	);
}
