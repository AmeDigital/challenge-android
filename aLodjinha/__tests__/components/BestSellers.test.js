import React from 'react';

import renderer from 'react-test-renderer';

import BestSellers from '../../src/components/BestSellers';

describe('Component BestSellers', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<BestSellers />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
