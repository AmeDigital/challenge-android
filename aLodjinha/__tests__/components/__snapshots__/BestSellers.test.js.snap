// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BestSellers should render without erros 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#aaaaaa",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2d3142",
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
      Mais Vendidos
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#5e2a84"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
